homeassistant/components/blink/config_flow.py

```diff
@@ -0,0 +1,99 @@
+"""Config flow to configure Blink."""
+import logging
+
+from blinkpy import blinkpy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_PASSWORD,
+    CONF_PIN,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+
+from .const import DEFAULT_OFFSET, DEFAULT_SCAN_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BlinkConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Blink config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize the blink flow."""
+        self.blink = None
+        self.data = {
+            CONF_USERNAME: "",
+            CONF_PASSWORD: "",
+            CONF_SCAN_INTERVAL: DEFAULT_SCAN_INTERVAL,
+        }
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if user_input is not None:
+            self.data[CONF_USERNAME] = user_input["username"]
+            self.data[CONF_PASSWORD] = user_input["password"]
+
+            await self.async_set_unique_id(self.data[CONF_USERNAME])
+
+            if CONF_SCAN_INTERVAL in user_input:
+                self.data[CONF_SCAN_INTERVAL] = user_input["scan_interval"]
+
+            try:
+                self.blink = self.hass.data[DOMAIN]
+            except KeyError:
+                self.blink = blinkpy.Blink(
+                    username=self.data[CONF_USERNAME],
+                    password=self.data[CONF_PASSWORD],
+                    motion_interval=DEFAULT_OFFSET,
+                    legacy_subdomain=False,
+                    no_prompt=True,
+                    device_id="Home Assistant",
+                )
+                self.blink.refresh_rate = self.data[CONF_SCAN_INTERVAL]
+                await self.hass.async_add_executor_job(self.blink.start)
+
+            if not self.blink.key_required:
+                # No key required, we're good
+                self.hass.data[DOMAIN] = {self.unique_id: self.blink}
```

Comment:
unique ID and config entry ID are not the same.

Reply:
the proper way is to drop the object and make the config entry create a new one. It's only once so not the end of the world.